packages/kit/src/runtime/client/client.js

```diff
@@ -1376,6 +1376,19 @@ export function create_client(app, target) {
 		},
 
 		goto: (href, opts = {}) => {
+			if (typeof href === 'string') {
+				href = new URL(href, get_base_uri(document));
```

Comment:
there are three places we do `new URL(href, get_base_uri(document))`. is that worthy of a helper?

Reply:
I only see two — where's the third?

as an aside the `if (typeof href === 'string')` check is arguably unnecessary, since if `href` (which is a badly named argument) is already a URL then it'll just be cloned harmlessly. `goto` isn't a hot code path so it's probably more valuable to shrink the code than to avoid the extra work

Reply:
one is `new URL(url, get_base_uri(document))`, but same thing. It also has that `if (typeof href === 'string')` check so we could either move it into the helper or remove it. I'm fine with removing it